net/blockwise/blockwise.go

```diff
@@ -889,6 +894,7 @@ func (b *BlockWise) processReceivedMessage(w ResponseWriter, r Message, maxSzx S
 	szx = getSzx(szx, maxSzx)
 	sendMessage := b.acquireMessage(r.Context())
 	sendMessage.SetToken(token)
+	// TODO What to set type to?
```

Comment:
In `v2` this is handled by:
https://github.com/plgd-dev/go-coap/blob/e59fa916285a183e0adbcbf6d0c2e1329ddc448e/udp/client/clientconn.go#L220
and
https://github.com/plgd-dev/go-coap/blob/e59fa916285a183e0adbcbf6d0c2e1329ddc448e/udp/client/clientconn.go#L296

for `v3` this will be refactored to block-wise.

Reply:
Hmm, I don't really get this, how is it handled by those lines, can you help me understand? I mean eventually the execution reaches this code

https://github.com/plgd-dev/go-coap/blob/948a0bcf18a66f92d7269f40104a64e997498d3b/udp/client/clientconn.go#L233 

and the message type is used to determine if there should be retries or not. So unless the type is set explicitly somewhere it will be random, `Confirmable` if it is a new message created in the message pool or it will be `NonConfirmable` if it is a reused message from the message pool.